qemu/tests/libvirt_host_model_test.py

```diff
@@ -0,0 +1,160 @@
+        virsh.start(name)
+    except Exception as err:
+        logging.error(err)
+        return
+    else:
+        qemu_cpu_info = cpu.get_cpu_info_from_virsh_qemu_cli(name)
+        libvirt_cpu_info = get_cpu_info_from_dumpxml(name)
+        params["qemu_cpu_info"] = qemu_cpu_info
+        params["libvirt_cpu_info"] = libvirt_cpu_info
+
+        cpu_model_qemu = qemu_cpu_info["model"]
+        cpu_model_libvirt = libvirt_cpu_info["model"]
+        if cpu_model_qemu != cpu_model_libvirt:
+            logging.error("mismatch cpu model bwteen qemu %s and libvirt %s",
+                          cpu_model_qemu, cpu_model_libvirt)
+            return False
+        qemu_cpu_flags = cpu.parse_qemu_cpu_flags(qemu_cpu_info['flags'])
+        libvirt_cpu_flags = libvirt_cpu_info['features']
+        qemu_cpu_flags = extend_flags_patterns(qemu_cpu_flags)
+        miss_flags = []
+        mismatch_flags = []
+        for flag in libvirt_cpu_flags.keys():
+            if flag not in qemu_cpu_flags.keys():
+                if libvirt_cpu_flags[flag] == 'on':
+                    miss_flags.append(flag)
+            elif libvirt_cpu_flags[flag] != qemu_cpu_flags[flag]:
+                mismatch_flags.append(flag)
+        if miss_flags:
+            logging.error("\nmiss flags %s from qemu cli\n", miss_flags)
+            if mismatch_flags:
+                logging.error("\nmismatch flags %s between libvirt and qemu\n",
+                              mismatch_flags)
+                return False
+            return False
+        return True
+    finally:
+        if virsh.is_alive(name):
+            virsh.destroy(name, ignore_status=True)
+        virsh.undefine(name, ignore_status=True)
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Test cpu models with full flags that libvirt generates
+    1) start a default configuration xml by virsh
+    2) get the cpu elements from libvirt xml by 'virsh dumpxml $domain'
+    3) parse the cpu_flags and store into libvirt_cpu_flags
+    4) get the cpu flags from qemu process and stroe to qemu_cpu_flags
+    5) Compare compare the cpu flags of xml and qemu process, should be same
+    6) Boot guest with qemu_cpu_flags, guest works well
+
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment.
+    """
+    if not compare_cpu_info(test, params):
+        logging.error("\ncpu libvirt flags are %s\n",
+                      (str(params["libvirt_cpu_info"])))
+        logging.error("\ncpu qemu cli flags are %s\n",
+                      (str(params["qemu_cpu_info"])))
+        test.fail("CPU info is different between dumpxml and qemu cli")
+
+    params["start_vm"] = "yes"
+    vm_name = params['main_vm']
+    env_process.preprocess_vm(test, params, env, vm_name)
+
+    vm = env.get_vm(vm_name)
+    error_context.context("Try to log into guest", logging.info)
+    session = vm.wait_for_login()
```

Comment:
Seems you will not use the session anymore, so can we remove it? Or if we can check the CPU flags in the guest as well?
```suggestion
    vm.wait_for_login()
```

Reply:
> Seems you will not use the session anymore, so can we remove it? Or if we can check the CPU flags in the guest as well?

yes, I just login guest without any other operation, update your suggestion, and thanks.